src/chapters/more-on-functions/functions-as-arguments.rst

```diff
@@ -0,0 +1,285 @@
+
+      function getValidInput(prompt, isValid) {
+         
+         let userInput = input.question(prompt);
+
+         while (!isValid(userInput)) {
+            console.log("Invalid input. Try again.");
+            userInput = input.question(prompt);
+         }
+
+         return userInput;
+      }
+
+      let isValidPassword = function(password) {
+
+         // Passwords should have at least 8 characters
+         if (password.length < 8) {
+            return false;
+         }
+
+         return true;
+      };
+
+      console.log(getValidInput('Create a password:', isValidPassword));
+
+   **Sample Output**
+
+   ::
+
+      Create a password: launch
+      Invalid input. Try again.
+      Create a password: code
+      Invalid input. Try again.
+      Create a password: launchcode
+      launchcode
+
+Example: A Logger
+^^^^^^^^^^^^^^^^^
+
+Another common example of a function using another function to customize its behavior is that of logging. A real-world application is typically capable of logging messages such as errors, warnings, and statuses. Many such applications allow for log messages to be sent to one or more destinations. For example, the application may log messages to both the console and to a file.
+
+We can write a logging function that relies on a function parameter to determine the logging destination.
+
+.. admonition:: Example
+
+   The ``logError`` function outputs a standardized error message to a location determined by the parameter ``logger``.
+
+   .. sourcecode:: js
+   
+      let fileLogger = function(msg) {
+
+         // Put the message in a file
+
+      }
+
+      function logError(msg, logger) {
+         logger('ERROR: ' + msg);
+      }
+
+      logError('Something broke!', fileLogger);
+
+This example can be made even more powerful by enabling multiple loggers.
+
+.. admonition:: Example
+
+   The call to ``logError`` will log the message to both the console and a file.
+
+   .. sourcecode:: js
+   
+      let fileLogger = function(msg) {
+
+         // Put the message in a file
+
+      }
+
+      let consoleLogger = function(msg) {
+      
+         console.log(msg);
+      
+      }
+
+      function logError(msg, loggers) {
+
+         for (let i = 0; i < loggers.length; i++) {
+            logger[i]('ERROR: ' + msg);
+         }
+
+      }   
+
+      logError('Something broke!', [fileLogger, consoleLogger]);
+
+As with the validation example, these programs separate behaviors in a way that makes the code more flexible. To add or remove a logging destination, we can simply change the way that we call ``logError``. The code *inside* ``logError`` doesn't know or care about the specfics of how each logging function does it's job. It is concerned merely with creating the message string and passing it to each logger.
+
+When writing a function that uses one of its parameters as a function, things can go wrong since there is now way to *force* a programmer to pass a function.
```

Comment:
I converted this chunk to be its own section, titled "A Word of Caution." It felt weird to me to use "*Caution:*" over the warning admonition, while using the latter would have resulted in three admonitions in a row with not text in between (also not great). I think the heading sets off the content appropriately. 

What do you suggest as an alternative to "a programmer"? I avoided using "us" or "you" because I wanted to emphasize that we don't just write functions for ourselves, but for others to use too.

Reply:
The separate section works well.  As for avoiding the programmer, maybe we could focus on the behavior of the code/function itself.  Something like, "Whenever a function uses one of its parameters as a function, errors occur if something other than a function gets passed."  I'm not happy with using 'function' 3 times in the sentence, but maybe we can adjust from here.